app/helpers/sampleLabelHelpers.js

```diff
@@ -1,70 +1,47 @@
-
-                    if(entity) {
-                        return entity[args.propertyName];
-                    }
-                }
-            }
-
-            return "";
-        },
-        UrlPicker: function(value, scope, args) {
-
-            if(!args.propertyName) {
-                args = {propertyName: "name"}
-            }
-
-            var entity;
-
-            switch (value.type) {
-                case "content":
-                    if(value.typeData.contentId) {
-                        entity = scope.services.archetypeCacheService.getEntityById(scope, value.typeData.contentId, "Document");
-                    }
-                    break;
-
-                case "media":
-                    if(value.typeData.mediaId) {
-                        entity = scope.services.archetypeCacheService.getEntityById(scope, value.typeData.mediaId, "Media");
-                    }
-                    break;
-
-                case "url":
-                    return value.typeData.url;
-                    
-                default:
-                    break;
-
-            }
-
-            if(entity) {
-                return entity[args.propertyName];
-            }
-
-            return "";
-        },
-        Rte: function (value, scope, args) {
-
-            if(!args.contentLength) {
-                args = {contentLength: 50}
-            }
-
-            return $(value).text().substring(0, args.contentLength);
-        }
+//create a namespace (optional)
+var ArchetypeSampleLabelHelpers = {};
+
+//create a function
+//you will add it to your label template field as `{{ArchetypeSampleLabelHelpers.testPromise(someArchetypePropertyAlias)}}`
+ArchetypeSampleLabelHelpers.testPromise = function(value) {   
+    //you can inject services
+    return function ($timeout, archetypeCacheService) {        
+        //best to return a promise
+        //NOTE: $timeout returns a promise
+        return $timeout(function () {
+            return "As Promised: " + value;
+        }, 1000);
+    }
+}
+
+ArchetypeSampleLabelHelpers.testEntityPromise = function(value, scope, args) {
+    //hey look, args!
+    //{{ArchetypeSampleLabelHelpers.testEntityPromise(someArchetypePropertyAlias, {foo: 1})}}
+    console.log(args);
+    
+    return function ($q, entityResource) {    
+        var deferred = $q.defer();
+    
+        entityResource.getById(args.foo, 'document').then(function(entity) {
+            console.log("Hello from testEntityPromise");
+            console.log(entity);
+            deferred.resolve(entity.name);
+        });
+    
+        return deferred.promise;
+    }
+}
+
+ArchetypeSampleLabelHelpers.testEntityPromise2 = function(value) {          
+    return function ($q, archetypeCacheService) {    
+        var deferred = $q.defer();
+    
+        archetypeCacheService.getEntityById(1054, 'document').then(function(entity) {
```

Comment:
Should "1054" be hardcoded here? Unsure of what "testEntityPromise2" is supposed to do, so can't be sure if this is correct or not.

Reply:
Ill make it nicer. The intent was to show the cache service injected. 👍